apps/web/src/views/Farms/FarmsV4.tsx

```diff
@@ -1,25 +1,39 @@
 import { useTranslation } from '@pancakeswap/localization'
 import { Card, Tab, TabMenu } from '@pancakeswap/uikit'
 import Page from 'components/Layout/Page'
-import { PropsWithChildren, useState } from 'react'
-import { Header } from './v4/components/Header'
+import { PropsWithChildren, useMemo, useState } from 'react'
+import { Header, Pools } from './v4/components'
 
 export const FarmsV4: React.FC<PropsWithChildren> = () => {
   const { t } = useTranslation()
   const [tab, setTab] = useState(0)
   const onTabClick = (index: number) => setTab(index)
 
+  const tabsConfig = useMemo(() => {
+    return {
+      0: {
+        menu: (counts: number) => <Tab>{t('All Pools (%counts%)', { counts })}</Tab>,
+        page: () => <Pools />,
+      },
+      1: {
+        menu: (counts: number) => <Tab>{t('My Positions (%counts%)', { counts })}</Tab>,
+        page: () => <Card>My Positions</Card>,
+      },
+      2: {
+        menu: () => <Tab>{t('History')}</Tab>,
+        page: () => <Card>History</Card>,
+      },
+    }
+  }, [t])
+
   return (
     <>
       <Header />
-
       <Page>
         <TabMenu activeIndex={tab} onItemClick={onTabClick} isShowBorderBottom={false}>
-          <Tab>{t('All Pools (%counts%)', { counts: 1 })}</Tab>
-          <Tab>{t('My Positions (%counts%)', { counts: 1 })}</Tab>
-          <Tab>{t('History')}</Tab>
+          {Object.values(tabsConfig).map(({ menu }) => menu(1))}
         </TabMenu>
-        <Card>1</Card>
+        {tabsConfig[tab].page()}
```

Comment:
maybe we move the page component out of current component, like use dynamic import

Reply:
we can just change the tabsConfig.page to import(...) whenever we need, i note it down, not in the PR